features/topology-map/plugins/org.opennms.features.topology.plugins.topo.graphml/src/main/java/org/opennms/features/topology/plugins/topo/graphml/GraphMLTopologyProvider.java

```diff
@@ -70,14 +69,25 @@ private static TopologyProviderInfo createTopologyProviderInfo(GraphMLGraph grap
         return new DefaultTopologyProviderInfo(name, description);
     }
 
+    public enum VertexStatusProviderType {
+        NO_STATUS_PROVIDER,
+        DEFAULT_STATUS_PROVIDER,
+        SCRIPT_STATUS_PROVIDER,
+        PROPAGATE_STATUS_PROVIDER,
+    }
+
+    private final GraphMLMetaTopologyProvider metaTopologyProvider;
     private final int defaultSzl;
     private final String preferredLayout;
     private final FocusStrategy focusStrategy;
     private final List<String> focusIds;
-    private final Boolean requiresStatusProvider;
+    private final Object vertexStatusProvider;
 
-    public GraphMLTopologyProvider(GraphMLGraph graph, GraphMLServiceAccessor serviceAccessor) {
+    public GraphMLTopologyProvider(final GraphMLMetaTopologyProvider metaTopologyProvider,
```

Comment:
Please do not expose the full MetaTopologyProvider. We should introduce something new, e.g.
OppositeVertexAccessor which is exposed. (see below)